JASP-Engine/JASP/R/commonMachineLearningClassification.R

```diff
@@ -0,0 +1,803 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+.readDataClassificationAnalyses <- function(dataset, options){
```

Comment:
This function is identical to `.readDataRegressionAnalyses` up to a single if statement. Perhaps they can use the same function? Also, some methods may use categorical predictors differently than numeric ones (which you'd tell them by making the predictor a factor) but you read all data in with `columns.as.numeric`..